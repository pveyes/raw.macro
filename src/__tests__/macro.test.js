const path = require("path");
const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");
const prettier = require("prettier");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(result, { trailingComma: "es5" });
  },
  tests: {
    "no usage": `import raw from '../macro'`,
    "correct usage": `
      import raw from '../macro';

      const macro = raw('raw.macro');
      const md = raw('./fixtures/markdown.md');
      const js = raw('./fixtures/javascript.js');
    `,
  },
});
