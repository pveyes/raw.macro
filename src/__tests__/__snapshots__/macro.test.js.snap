// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros correct usage: correct usage 1`] = `
"
import raw from '../macro';

const macro = raw('raw.macro');
const md = raw('./fixtures/markdown.md');
const js = raw('./fixtures/javascript.js');

      ↓ ↓ ↓ ↓ ↓ ↓

const macro =
  '/* istanbul ignore next */\\\\nimport macro from \\"./macro\\";\\\\n\\\\nexport default macro;\\\\n';
const md = \\"# This is the title\\\\n\\\\nThis is the description\\\\n\\";
const js =
  'const sum = require(\\"../sum\\");\\\\n\\\\ntest(\\"add two numbers\\", () => {\\\\n  const result = sum(1, 2);\\\\n  expect(result).toEqual(3);\\\\n});\\\\n';

"
`;

exports[`macros no usage: no usage 1`] = `
"
import raw from '../macro'

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;
